boot/bootutil/src/loader.c

```diff
@@ -386,13 +387,14 @@ boot_write_status(const struct boot_loader_state *state, struct boot_status *bs)
 /*
  * Validate image hash/signature and optionally the security counter in a slot.
  */
-static int
+static fih_int
 boot_image_check(struct boot_loader_state *state, struct image_header *hdr,
                  const struct flash_area *fap, struct boot_status *bs)
 {
     TARGET_STATIC uint8_t tmpbuf[BOOT_TMPBUF_SZ];
     uint8_t image_index;
-    int rc;
+    int rc = -1;
+    fih_int fih_rc = FIH_FAILURE;
```

Comment:
```suggestion
    fih_int fih_rc ;
```
always assigned by FIH_CALL(bootutil_img_validate...

Reply:
It is intended to protect against attack when the PC is tried to abuse such a way (manipulating lower bits with EM pulse) to jump at the end of the function and just returns without performing any operation. Very unlikely and hard to achieve in practice. But during a certification process, code reviewer can mandate to have negative return value set by default.

Reply:
Yes, all variables should be initialized to the error value, otherwise abuse and unintended early exit would result in success returns. I wonder if those could not just be initialized at the beginning of the function instead of where they were declared, but I don't have strong opinions about it anymore.

Reply:
Ok, I'm fine with this clarification - although all that might be compiled out as optimization.